packages/devtools_app/lib/src/timeline/flutter/timeline_processor.dart

```diff
@@ -0,0 +1,615 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:meta/meta.dart';
+
+import '../../config_specific/logger/logger.dart';
+import '../../ui/fake_flutter/fake_flutter.dart';
+import '../../utils.dart';
+//import '../simple_trace_example.dart';
+import 'timeline_controller.dart';
+import 'timeline_model.dart';
+
+// For documentation, see the Chrome "Trace Event Format" document:
+// https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU.
+// This class depends on the stability of event names we receive from the
+// engine. That dependency is tracked at
+// https://github.com/flutter/flutter/issues/27609.
+
+/// Switch this flag to true collect debug info from the timeline protocol.
+///
+/// This will add a button to the timeline page that will download files with
+/// debug info on click.
+bool debugTimeline = false;
+
+/// List that will store trace event json in the order we handle the events.
+///
+/// This list is for debug purposes. When [debugTimeline] is true, we will be
+/// able to dump this data to a downloadable text file.
+List<Map<String, dynamic>> debugHandledTraceEvents = [];
+
+/// Buffer that will store significant events in the frame tracking process.
+///
+/// This buffer is for debug purposes. When [debugTimeline] is true, we will
+/// be able to dump this buffer to a downloadable text file.
+StringBuffer debugFrameTracking = StringBuffer();
+
+const String gpuEventName = 'GPURasterizer::Draw';
+
+const String uiEventName = 'VSYNC';
+
+/// Processor for composing a recorded list of trace events into a timeline of
+/// [AsyncTimelineEvent]s, [SyncTimelineEvent]s, and [TimelineFrame]s.
+class TimelineProcessor {
```

Comment:
In this file I combined the logic that was previously split between `FrameBasedTimelineProcessor` and `FullTimelineProcessor`